### ISSUE TITLE
New `addon` Hook

### ISSUE THREAD

**Issue by jhunt** [starkandwayne/genesis#258]
Implement support for a new `addon` hook that allows Genesis kits to execute arbitrary commands, as pre-packaged scripts, when a user calls them thusly:

```
genesis do <env> <script-name> [args...]
```

Examples we have come up with so far:

```
# bosh kit
genesis do jhunt-lab login
genesis do jhunt-lab smoke-tests
genesis do jhunt-lab upload-stemcells
genesis do jhunt-lab aws-upload-keys

# cf kit
genesis do jhunt-lab login
genesis do jhunt-lab smoke-tests
```

I'm sure there's tons more.

This is a great way to handle arbitrary complicated logic without mucking up Genesis internals with kit-specific code paths.

**Comment by dennisjbell** [comment]
Does this not duplicate #143?  If not, how does it differ?

**Comment by jhunt** [comment]
It's the same spirit as #143, different implementation.

**Comment by jhunt** [comment]
This has been implemented int he v2.6 line.
